pkg/repository/provider/unified_repo.go

```diff
@@ -419,14 +419,16 @@ func getStorageCredentials(backupLocation *velerov1api.BackupStorageLocation, cr
 	switch backendType {
 	case repoconfig.AWSBackend:
 		credValue, err := getS3Credentials(config)
-		if err != nil {
+		if err == nil {
+			result[udmrepo.StoreOptionS3KeyID] = credValue.AccessKeyID
+			result[udmrepo.StoreOptionS3Provider] = credValue.ProviderName
+			result[udmrepo.StoreOptionS3SecretKey] = credValue.SecretAccessKey
+			result[udmrepo.StoreOptionS3Token] = credValue.SessionToken
+		} else if err == repoconfig.AWSCredFileMissingError {
+			log.Warn("AWS credential file is not present, so no credentials are provided")
```

Comment:
If this happens, then there is no credential information in the result, but the IAM way also mounts the credential information into the Velero pod.
Is there a place where the Kopia Library can read credential information?

Reply:
+1
Also curious how kopia handles different auth mechanisms.

Reply:
What Velero needs is letting it go without checking static ak/sk as well.
For sure, the IAM role SA should be mounted to Velero pods, this has been done by another PR https://github.com/vmware-tanzu/velero/pull/5802

Reply:
Moreover, it looks weird to let it go without checking anything when getting credentials, so here I have modified the code to check the existence of the env variable AWS_ROLE_ARN, only when its value is not empty, we let it go with empty ak/sk